terminology.md

```diff
@@ -261,17 +261,23 @@ A realm consists of a set of [intrinsic](#Intrinsic) objects, an ECMAScript glob
 
 ### Intrinsic
 #### Definition
-Built-in objects that are explicitly referenced by the algorithms of the ECMA262 specification.
+Built-in objects (usually realm-specific) that are explicitly referenced by the algorithms of the ECMA262 specification.
 #### Example
-`[]`
+%ForInIteratorPrototype%
 #### Reference
 [ECMA262 Spec](https://tc39.es/ecma262/#sec-well-known-intrinsic-objects)
 
 ### Primordial
 #### Definition
-Global, mutable objects that must exist before any code runs.
+The subset of [intrinsic](#Intrinsic) objects that are accessible to ECMAScript code.
```

Comment:
The definition of "primordial" (or by the truce with @allenwb a "primordial intrinsic") is an object that is mandated by ECMA262 to exist before code starts running. Certainly all primordials are accessible to ECMAScript code. IIUC, so are all intrinsics. The "objects" in the spec that are not accessible to ECMAScript code are internal something (don't remember the terminology) but they aren't really objects. They are internal spec fictions used to explain the behaviors of things that are observable.

Reply:
> an object that is mandated by ECMA262 to exist before code starts running

AFAICT, whether or not an object exists before code starts running is neither observable by ECMAScript code nor mandated by spec. To give a concrete example, I believe a conforming implementation can defer creation of Date (and in particular determination of local time zone adjustment) until just before the absence would be observable and lock in local time zone adjustment at that time, such that changing the host configuration could affect running ECMAScript code only if that code has not yet encountered such a case.

> Certainly all primordials are accessible to ECMAScript code. IIUC, so are all intrinsics

%ForInIteratorPrototype% is an intrinsic that is not accessible to ECMAScript code: https://tc39.es/ecma262/#sec-%foriniteratorprototype%-object
These changes attempt to clarify that Undeniables ⊊ Primordials ⊊ Intrinsics, specifically that each of those relations are strict and the sets are therefore distinct. %ForInIteratorPrototype% is Intrinsic but not Primordial, and %Array% is Primordial but not Undeniable.